diff --git a/libafl_sugar/src/forkserver.rs b/libafl_sugar/src/forkserver.rs
index 2b863c2..d694998 100644
--- a/libafl_sugar/src/forkserver.rs
+++ b/libafl_sugar/src/forkserver.rs
@@ -31,7 +31,7 @@ use libafl::{
 
 use crate::{CORPUS_CACHE_SIZE, DEFAULT_TIMEOUT_SECS};
 
-pub const MAP_SIZE: usize = 65536;
+pub const MAP_SIZE: usize = 80642;
 
 #[derive(TypedBuilder)]
 pub struct ForkserverBytesCoverageSugar<'a> {
