diff --git a/libafl/src/executors/forkserver.rs b/libafl/src/executors/forkserver.rs
index 2340b0c..fec3c77 100644
--- a/libafl/src/executors/forkserver.rs
+++ b/libafl/src/executors/forkserver.rs
@@ -372,9 +372,24 @@ where
                     .copy_from_slice(input.target_bytes().as_slice());
             }
             None => {
+                let mut immutable_header = vec![
+                    // pcap header
+                    0xd4, 0xc3, 0xb2, 0xa1, 0x02, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    0x00, 0x00, 0x00, 0xff, 0xff, 0x00, 0x00, 0xe4, 0x00, 0x00, 0x00, 0xb7, 0xc8,
+                    0x9e, 0x61, 0x3b, 0x35, 0x0e, 0x00, 0x08, 0x01, 0x00, 0x00, 0x08, 0x01, 0x00,
+                    0x00, // ip header next
+                    0x45, 0x00, 0x01, 0x08, 0x00, 0x01, 0x00, 0x00, 0x40, 0x11, 0x7a, 0xe1, 0x7f,
+                    0x00, 0x00, 0x01, 0x7f, 0x01, 0x01, 0x01, // udp header next
+                    0x00, 0x43, 0x00, 0x44, 0x00, 0xf4, 0xf7,
+                    0x7a,
+                    // everything after is bootp
+                ];
+
+                immutable_header.extend_from_slice(&input.target_bytes().as_slice());
+
                 self.executor
                     .out_file_mut()
-                    .write_buf(input.target_bytes().as_slice());
+                    .write_buf(immutable_header.as_slice());
             }
         }
 
